app/src/main/java/com/github/se/cyrcle/ui/authentication/Authenticator.kt

```diff
@@ -0,0 +1,108 @@
+package com.github.se.cyrcle.ui.authentication
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.github.se.cyrcle.R
+import com.github.se.cyrcle.model.user.User
+import com.github.se.cyrcle.ui.theme.ColorLevel
+import com.github.se.cyrcle.ui.theme.atoms.Text
+import com.github.se.cyrcle.ui.theme.googleSignInButtonStyle
+
+/**
+ * Interface for a Composable repository It provides buttons with callbacks once authentication is
+ * complete This does not interface with the UserRepository or ViewModel
+ */
+interface Authenticator {
+
+  /**
+   * Composable button that authenticates the user
+   *
+   * @param onAuthComplete callback for when authentication is successful
+   * @param onAuthError callback for when authentication fails
+   */
+  @Composable
+  fun AuthenticateButton(onAuthComplete: (User) -> Unit, onAuthError: (Exception) -> Unit)
+
+  /**
+   * Composable button that signs the user in anonymously
+   *
+   * @param onComplete callback for when sign in is complete
+   */
+  @Composable fun SignInAnonymouslyButton(onComplete: () -> Unit)
+
+  /** Static methods for default composable buttons */
+  companion object {
+
+    /**
+     * Default composable button for authentication
+     *
+     * @param onClick callback for when the button is clicked
+     */
+    @Composable
+    fun DefaultAuthenticateButton(onClick: () -> Unit) {
+      Button(
+          onClick = onClick,
+          colors = ButtonDefaults.buttonColors(containerColor = Color.White),
```

Comment:
Careful with using hard-coded colors value.

Reply:
It's the template button used in all apps, it should be ok 👌